src/path.c

```diff
@@ -1,90 +1,114 @@
 #include "path.h"
 
-static bool straight = true;
-static char *source;
-static enum movement *destination;
+struct translation {
+	char *from;
+	enum movement to;
+};
 
-/**
- * @brief Translate the source path when coming from a straight movement.
- *
- * @return Number of movements translated from source.
- */
-static int from_straight(void)
-{
-	if (!strncmp(source, "F", 1)) {
-		*destination = MOVE_FRONT;
-		return 1;
-	}
-	if (!strncmp(source, "BF", 2)) {
-		*destination = MOVE_START;
-		return 1;
-	}
-	if (!strncmp(source, "LF", 2)) {
-		*destination = MOVE_LEFT_90;
-		return 1;
-	}
-	if (!strncmp(source, "RF", 2)) {
-		*destination = MOVE_RIGHT_90;
-		return 1;
-	}
-	if (!strncmp(source, "LLF", 3)) {
-		*destination = MOVE_LEFT_180;
-		return 2;
-	}
-	if (!strncmp(source, "RRF", 3)) {
-		*destination = MOVE_RIGHT_180;
-		return 2;
-	}
-	straight = false;
-	return 0;
-}
+struct dictionary {
+	struct translation *from_straight;
+	struct translation *from_diagonal;
+};
+
```

Comment:
I used `.from_diagonal` on `PATH_SAFE` because the movement is a diagonal (even if we avoid it and execute consecutive turns instead).

Yeah, added a comment in: 80d54777c1c6aef2907dc70edca0524c63812db4

:blush:

Reply:
great